README
Carlos Augusto Monterrosa Lopez / Rafael Hernandez Escobar
cmonterrosa@gmail.com - snakheertax@gmail.com
http://cmonterrosatechnotes.blogspot.com
=== SISCCO - Sistema de Credito y Cobranza para Instituciones Financieras 1.0 ===
México - Agosto de 2010
Technology: Ruby on Rails/HTML/CSS/AJAX/JasperReports


CHANGELOG
Modificaciones 2 de abril de 2012

=> creacion de restricciones fisicas en la BD


Modificaciones 16 de febrero de 2011

=> Corregimos la carga de layouts, ya cargan todos y se almacenan en una tabla temporal pendientes de aplicar

=> Agregamos indices para mejorar consulta por curp o apellido paterno




Modificaciones 8 de febrero de 2011

=> Corrección de la exportacion de la información de creditos exportada en csv (agregamos localidad, municipio y num_socias)
   se encuentra dentro del menú principal en el apartado de reportes

=> Se cargo al sistema la imagen de logo de cresolido

=> Se agrego el reporte "recibo de entrega de fondos en pdf"

=> Se modifico el reporte "carta compromiso de acuerdo al formato dado"

=> Se modifico función que genera la exportación del tarjeton de pagos individual




Modificaciones 15 de diciembre de 2010

=> Reajustamos la funcion que calcula el moratorio

=> Reajustamos la funcion de capital vencido

=> Corregimos mensaje que enviaba al cargar nuevamente el layout

=> Se modifico la funcion que calcula los dias de vencimiento sin importar que se incremente el año




Modificaciones 25 de noviembre de 2010

=> Optimizamos tamaño de las imagenes para que cargen mas rapido

=> Optimizamos consultas de clientes a la Base de datos

=> Implentamos la busqueda por CURP en los clientes


Modificamos 24 de noviembre de 2010

=>  Creamos funcion que confronta la referencia que recibe del layout con la referencia del credito, si esta es valida inserta el deposito, queda pendiente aplicarlo

=> Creamos funcion que despliega los que se aplicaron y los que no pudieron ser aplicados, se descargan en un archivo de texto, de igual manera se descarga otro archivo si algun error ocurre

=> Validamos la estructura del layout para que unicamente cuando sea correcta pueda procesarse

=> Por seguridad, Validamos que no pueda volverse a subir 2 veces el mismo archivo

=> Vinculamos el deposito con un credito, para acceder más rapido a él

=> En el menu de crédito mostramos todos los depositos que se han recibido via layout



Modificaciones 18 de noviembre de 2010

=> Corregimos, al crear productos o fondeos, estos aparecen inmediatamente en la alta de crédito grupal

=> Agregamos el botón de "SALDO" para verificar el saldo, al hacer la búsqueda del crédito

=> Agregamos el botón "Calendario de Pagos"; para conocer las fechas de pago

=> Optimizamos variables para hacer má rápidas las consultas

=>Definimos esquema para leer el layout


Modificaciones 4 de noviembre de 2010

=> Creamos el acceso para que se calcule el vencimiento por credito especifico

=> Creamos el reporte que calculo el vencimiento general de todos los creditos


Modificaciones 29 de Octubre de 2010

=> Creamos clase para calcular los vencimientos

=> Optimizamos variables globales


Modificaciones 13 de Octubre de 2010

=> Quitar el boton de eliminar para los productos y lineas para evitar inconsitencia de informacion

=> Modificar la vista de clientes, mostrando unicamente los que no pertenecen a un grupo para ser agregados

=> En alta de creditos, filtrado de grupos unicamente los que tienen clientes activos



Modificaciones 3 de Octubre de 2010

=> Creamos la estructura en la Base de Datos para almacenar los devengos semanales y diarios segun sea el caso automaticemente

=> Generamos funcion que calcula de acuerdo al credito, los devengos, queda a la espera de hacerse un match
   para generarse las polizas

=> Es necesario que recibamos el catalogo de las cuentas contables para importarlas al sistema y empezar a
   vincularlo con los devengos y al generar un nuevo credito

=> Generamos la exportacion a excel de las polizas con la estructura del conta 2000

=> Generamos el acceso a visualizar los devengos por credito en el menu de busqueda por credito, id, grupo, etc..

=> Creamos el acceso al catalogo de cuentas contables

=> Quitamos el acceso al catalogo de polizas, lo hará internamente el sistema


Modificaciones 30 de septiembre de 2010

=> Agregamos al menu principal el acceso a sucursales bancarias

=> Modificamos el calculo de intereses moratorio para abarcar más de un año segun sea el caso

=> Se vinculo la linea de creditos con la cuenta liquidadora y concentradora

=> Se quito de banco, sucursal y num_cuenta de la alta de credito, ahora esta vinculado con la linea de credito

=> Corregimos titulos y nombres de botones

=> Mejoramos los combos adicionando la suma de varios campos (cta concentradora, cta liquidadora, suc bancarias)

=> Se agrego periodo catorcenal

=> Se agrego la visualizacion de los creditos con los que cuenta un promotor, desde el catalogo de promotores

=> Se agrego una funcion que no permite escribir mas de 18 caracteres para curp, 13 para rfc, 18 para la clave del ife y 5
   para el codigo postal en la alta de cliente


Modificaciones 29 de Septiembre del 2010
=> Se crearon los catalogos ctaconcentradoras, ctaliquidadoras, sucbancarias, se modificó el catalogo de bancos,
   el catalogo de lineas, y se creo un submenú para los catalogos de ctas concentradoras y liquidadoras

Modificaciones 28 de septiembre de 2010

=> Modificamos la funcion que genera la llave alfanumerica, le quitamos el prefijo




Modificaciones 27 de septiembre de 2010

=> Agregamos la imagen y la funcionalidad para validar formato y existencia de curp, al dar de alta clientes

=> Optimizamos el servidor web, creando un cluster para balancerar la carga del procesamiento

=> Agregamos algunos indices (curp, nombre, etc) para optimizar las consultas a la base de datos

=> Implementamos un programa que realiza diariamente el respaldo automatico y compacta la base de datos




Modificaciones 23 de septiembre de 2010

=> Corregimos duplicidad del fondeo

=> Corregimos el eliminar o consultar a través del filtro grupo cuando se hace la busqueda


db/schema.rb
# This file is autogenerated. Instead of editing this file, please use the
# migrations feature of ActiveRecord to incrementally modify your database, and
# then regenerate this schema definition.

ActiveRecord::Schema.define(:version => 64) do

  create_table "actividads", :force => true do |t|
    t.column "clave_inegi", :string
    t.column "actividad",   :string
  end

  create_table "ahorros", :force => true do |t|
    t.column "monto",      :string
    t.column "cliente_id", :integer
    t.column "credito_id", :integer
    t.column "hora_fecha", :datetime
  end

  create_table "bancos", :force => true do |t|
    t.column "nombre", :string
  end

  create_table "civils", :force => true do |t|
    t.column "civil", :string
  end

  create_table "clientes", :force => true do |t|
    t.column "identificador",   :string,  :limit => 18
    t.column "paterno",         :string
    t.column "materno",         :string
    t.column "nombre",          :string
    t.column "fecha_nac",       :date
    t.column "rfc",             :string,  :limit => 13
    t.column "curp",            :string,  :limit => 18
    t.column "clave_ife",       :string
    t.column "sexo",            :string,  :limit => 1
    t.column "tipo_propiedad",  :string
    t.column "tipo_persona",    :string
    t.column "direccion",       :string
    t.column "colonia",         :string
    t.column "codigo_postal",   :string
    t.column "telefono",        :string,  :limit => 10
    t.column "fax",             :string,  :limit => 10
    t.column "email",           :string
    t.column "folio_rfc",       :string,  :limit => 13
    t.column "civil_id",        :integer
    t.column "escolaridad_id",  :integer
    t.column "vivienda_id",     :integer
    t.column "localidad_id",    :integer
    t.column "nacionalidad_id", :integer
    t.column "edo_residencia",  :string
    t.column "st",              :integer
    t.column "num_exterior",    :string,  :limit => 10
    t.column "num_interior",    :string,  :limit => 10
    t.column "rol_hogar",       :string,  :limit => 50
  end

  add_index "clientes", ["paterno", "materno", "nombre"], :name => "clientes_nombrecompleto"

  create_table "clientes_grupos", :force => true do |t|
    t.column "cliente_id",   :integer
    t.column "grupo_id",     :integer
    t.column "activo",       :integer
    t.column "fecha_inicio", :datetime
    t.column "fecha_fin",    :datetime
  end

  add_index "clientes_grupos", ["cliente_id"], :name => "index_clientes_grupos_on_cliente_id"
  add_index "clientes_grupos", ["grupo_id"], :name => "index_clientes_grupos_on_grupo_id"

  create_table "colonias", :force => true do |t|
    t.column "colonia",     :string
    t.column "clave_inegi", :string
    t.column "ejido_id",    :integer
    t.column "st",          :integer
  end

  create_table "configuracion", :force => true do |t|
    t.column "nombre_empresa",    :string
    t.column "direccion",         :string
    t.column "ciudad",            :string
    t.column "prefijo",           :string
    t.column "telefono",          :string
    t.column "activo",            :string,  :default => "1"
    t.column "ultima_referencia", :integer
    t.column "ultima_poliza",     :integer
  end

  create_table "controllers", :force => true do |t|
    t.column "controller",  :string
    t.column "descripcion", :string
  end

  create_table "creditos", :force => true do |t|
    t.column "fecha_inicio",      :date
    t.column "fecha_fin",         :date
    t.column "num_referencia",    :string
    t.column "monto",             :float
    t.column "tasa_interes",      :float
    t.column "interes_moratorio", :string
    t.column "identificador",     :string
    t.column "tipo_interes",      :string
    t.column "linea_id",          :integer
    t.column "cliente_id",        :integer
    t.column "promotor_id",       :integer
    t.column "destino_id",        :integer
    t.column "grupo_id",          :integer
    t.column "producto_id",       :integer
    t.column "status",            :integer
    t.column "fecha_captura",     :date
    t.column "tipo_aplicacion",   :string
    t.column "monto_inicial",     :decimal,  :precision => 15, :scale => 10
    t.column "updated_at",        :datetime
    t.column "created_at",        :datetime
  end

  create_table "ctaconcentradoras", :force => true do |t|
    t.column "num_cta",        :string
    t.column "cuenta_id",      :integer
    t.column "sucbancaria_id", :integer
  end

  create_table "ctaliquidadoras", :force => true do |t|
    t.column "num_cta",        :string
    t.column "cuenta_id",      :integer
    t.column "sucbancaria_id", :integer
  end

  create_table "cuentas", :force => true do |t|
    t.column "sCtaNum", :string, :limit => 20
    t.column "sNombre", :string, :limit => 50
  end

  create_table "datafiles", :force => true do |t|
    t.column "clave",              :string,   :limit => 4
    t.column "numero",             :string,   :limit => 4
    t.column "numero_cliente",     :string,   :limit => 4
    t.column "nombre_cliente",     :string,   :limit => 120
    t.column "codigo",             :string,   :limit => 18
    t.column "fecha_hora_archivo", :datetime
    t.column "fecha_hora_carga",   :datetime
    t.column "sucursal",           :string,   :limit => 10
    t.column "cuenta",             :string,   :limit => 10
    t.column "nombre_cuenta",      :string,   :limit => 120
    t.column "nombre_archivo",     :string
    t.column "moneda",             :string,   :limit => 50
    t.column "num_movimientos",    :integer
  end

  create_table "depositos", :force => true do |t|
    t.column "sucursal",     :string,   :limit => 20
    t.column "autorizacion", :string,   :limit => 20
    t.column "codigo",       :string,   :limit => 20
    t.column "subcodigo",    :string,   :limit => 20
    t.column "ref_num",      :string,   :limit => 20
    t.column "ref_alfa",     :string,   :limit => 20
    t.column "importe",      :string,   :limit => 20
    t.column "st",           :string,   :limit => 2
    t.column "fecha_hora",   :datetime
    t.column "datafile_id",  :integer
    t.column "credito_id",   :integer
  end

  create_table "destinos", :force => true do |t|
    t.column "destino",    :string
    t.column "user_id",    :integer
    t.column "fecha_hora", :datetime
  end

  create_table "devengos", :force => true do |t|
    t.column "fecha",                   :date
    t.column "dia",                     :integer
    t.column "semana",                  :integer
    t.column "generacion_obligacion",   :string,  :limit => 18
    t.column "vto_amortizacion",        :string,  :limit => 18
    t.column "cuenta_contable_interes", :string,  :limit => 18
    t.column "credito_id",              :integer
  end

  create_table "ejidos", :force => true do |t|
    t.column "ejido",        :string
    t.column "clave_inegi",  :string
    t.column "municipio_id", :integer
    t.column "st",           :integer
  end

  create_table "escolaridads", :force => true do |t|
    t.column "escolaridad", :string
    t.column "estudios_id", :integer
  end

  create_table "estados", :force => true do |t|
    t.column "estado",     :string
    t.column "edo_inegi",  :string
    t.column "edo_renapo", :string
  end

  create_table "extraordinarios", :force => true do |t|
    t.column "credito_id",         :integer
    t.column "status",             :integer
    t.column "interes",            :decimal, :precision => 15, :scale => 10
    t.column "capital",            :decimal, :precision => 15, :scale => 10
    t.column "proporcion_capital", :decimal, :precision => 15, :scale => 10
    t.column "proporcion_interes", :decimal, :precision => 15, :scale => 10
  end

  create_table "fechavalors", :force => true do |t|
    t.column "fecha",        :date
    t.column "sucursal",     :string,   :limit => 20
    t.column "autorizacion", :string,   :limit => 20
    t.column "codigo",       :string,   :limit => 20
    t.column "subcodigo",    :string,   :limit => 20
    t.column "ref_alfa",     :string,   :limit => 20
    t.column "importe",      :string,   :limit => 20
    t.column "st",           :string,   :limit => 2
    t.column "fecha_hora",   :datetime
    t.column "datafile_id",  :integer
    t.column "credito_id",   :integer
  end

  create_table "festivos", :force => true do |t|
    t.column "fecha",       :date
    t.column "descripcion", :string
  end

  create_table "fondeos", :force => true do |t|
    t.column "acronimo",  :string
    t.column "fuente",    :string
    t.column "domicilio", :string
    t.column "telefono",  :string
  end

  create_table "garantias", :force => true do |t|
    t.column "garantia", :string
  end

  create_table "garantias_referencias", :id => false, :force => true do |t|
    t.column "referencia_id", :integer
    t.column "garantia_id",   :integer
  end

  create_table "giros", :force => true do |t|
    t.column "giro",      :string
    t.column "codigo",    :string
    t.column "subsector", :string
  end

  create_table "grupos", :force => true do |t|
    t.column "nombre", :string
  end

  create_table "jerarquias", :force => true do |t|
    t.column "jerarquia", :string
  end

  create_table "lineas", :force => true do |t|
    t.column "cuenta_cheques",      :string
    t.column "fecha_aut",           :date
    t.column "autorizado",          :float
    t.column "estatus",             :string,  :limit => 11
    t.column "gcnf",                :string
    t.column "fondeo_id",           :integer
    t.column "ctaconcentradora_id", :integer
    t.column "ctaliquidadora_id",   :integer
  end

  create_table "localidads", :force => true do |t|
    t.column "loc_id",       :string
    t.column "municipio_id", :integer
    t.column "localidad",    :string
  end

  create_table "logs", :force => true do |t|
    t.column "operacion",  :string
    t.column "fecha_hora", :datetime
    t.column "clase",      :string
    t.column "user_id",    :integer
    t.column "objeto_id",  :integer
  end

  create_table "miembros", :force => true do |t|
    t.column "credito_id",   :integer
    t.column "jerarquia_id", :integer
    t.column "cliente_id",   :integer
  end

  create_table "movimientos", :force => true do |t|
    t.column "tipo",     :string
    t.column "capital",  :float
    t.column "fecha",    :date
    t.column "interes",  :float
    t.column "concepto", :string
    t.column "pago_id",  :integer
  end

  create_table "municipios", :force => true do |t|
    t.column "municipio",   :string
    t.column "clave_inegi", :string
    t.column "estado_id",   :integer
  end

  create_table "nacionalidads", :force => true do |t|
    t.column "pais_id",   :string
    t.column "pais",      :string
    t.column "pais_gent", :string
  end

  create_table "negocios", :force => true do |t|
    t.column "nombre",               :string
    t.column "puesto",               :string
    t.column "direccion",            :string
    t.column "descripcion",          :string
    t.column "telefono",             :string,  :limit => 10
    t.column "num_empleados",        :integer
    t.column "ing_semanal",          :float
    t.column "cliente_id",           :integer
    t.column "actividad_id",         :integer
    t.column "ubicacion_negocio_id", :integer
  end

  create_table "pagogrupals", :force => true do |t|
    t.column "fecha_limite",         :date
    t.column "capital_minimo",       :float
    t.column "interes_minimo",       :float
    t.column "pagado",               :integer
    t.column "credito_id",           :integer
    t.column "num_pago",             :integer
    t.column "saldo_inicial",        :float
    t.column "saldo_final",          :float
    t.column "principal_recuperado", :float
  end

  create_table "pagoextraordinarios", :force => true do |t|
    t.column "extraordinario_id", :integer
    t.column "fecha",             :date
    t.column "cantidad",          :decimal, :precision => 15, :scale => 10
  end

  create_table "pagogrupals", :force => true do |t|
    t.column "fecha_limite",         :date
    t.column "capital_minimo",       :float
    t.column "interes_minimo",       :float
    t.column "pagado",               :integer
    t.column "credito_id",           :integer
    t.column "num_pago",             :integer
    t.column "saldo_inicial",        :float
    t.column "saldo_final",          :float
    t.column "principal_recuperado", :float
  end

  create_table "pagos", :force => true do |t|
    t.column "num_pago",             :integer
    t.column "fecha_limite",         :date
    t.column "capital_minimo",       :string
    t.column "interes_minimo",       :string
    t.column "moratorio",            :string
    t.column "pagado",               :integer
    t.column "credito_id",           :integer
    t.column "cliente_id",           :integer
    t.column "descripcion",          :string
    t.column "int_devengados",       :float
    t.column "saldo_inicial",        :string
    t.column "saldo_final",          :string
    t.column "principal_recuperado", :string
  end

  create_table "pagoslineas", :force => true do |t|
    t.column "linea_id",      :integer
    t.column "fecha",         :date
    t.column "monto",         :string
    t.column "observaciones", :string
    t.column "st",            :integer
  end

  create_table "periodos", :force => true do |t|
    t.column "nombre",    :string
    t.column "dias",      :integer
    t.column "pagos_mes", :integer
  end

  create_table "polizas", :force => true do |t|
    t.column "fecha",         :date
    t.column "tipo_poliza",   :string,  :limit => 1
    t.column "num_poliza",    :string,  :limit => 6
    t.column "cuenta_id",     :integer
    t.column "naturaleza",    :string,  :limit => 25
    t.column "importe",       :string,  :limit => 18
    t.column "descripcion",   :string,  :limit => 60
    t.column "identificador", :string,  :limit => 3
  end

  create_table "productos", :force => true do |t|
    t.column "producto",          :string,  :limit => 100
    t.column "intereses",         :float
    t.column "moratorio_ssi",     :float
    t.column "ahorro",            :float
    t.column "num_pagos",         :integer
    t.column "tasa_anualizada",   :string,  :limit => 10
    t.column "periodo_id",        :integer
    t.column "moratorio_flat",    :float
    t.column "tasa_mensual_flat", :float
    t.column "iva",               :decimal,                :precision => 8, :scale => 2
    t.column "gastos_cobranza",   :decimal,                :precision => 8, :scale => 2
  end

  create_table "promotors", :force => true do |t|
    t.column "paterno",       :string
    t.column "materno",       :string
    t.column "nombre",        :string
    t.column "direccion",     :string
    t.column "telefono",      :string,  :limit => 10
    t.column "celular",       :string,  :limit => 10
    t.column "email",         :string
    t.column "observaciones", :string
    t.column "sucursal_id",   :integer
    t.column "st",            :integer
  end

  create_table "referencias", :force => true do |t|
    t.column "paterno",    :string
    t.column "materno",    :string
    t.column "nombre",     :string
    t.column "parentesco", :string
    t.column "direccion",  :string
    t.column "telefono",   :string,  :limit => 10
    t.column "tipo",       :string
    t.column "credito_id", :integer
  end

  create_table "rol_hogars", :force => true do |t|
    t.column "rol",   :string
    t.column "alias", :string
  end

  create_table "roles", :force => true do |t|
    t.column "name", :string
  end

  create_table "roles_users", :id => false, :force => true do |t|
    t.column "user_id", :integer
    t.column "role_id", :integer
  end

  add_index "roles_users", ["role_id"], :name => "index_roles_users_on_role_id"
  add_index "roles_users", ["user_id"], :name => "index_roles_users_on_user_id"

  create_table "rols", :force => true do |t|
    t.column "nombre", :string
    t.column "st",     :integer
  end

  create_table "sectors", :force => true do |t|
    t.column "sector", :string
  end

  create_table "status", :force => true do |t|
    t.column "statu", :string
  end

  create_table "subsectors", :force => true do |t|
    t.column "subsector", :string
    t.column "sector_id", :integer
  end

  create_table "sucbancarias", :force => true do |t|
    t.column "num_sucursal", :string,  :limit => 4
    t.column "nombre",       :string
    t.column "banco_id",     :integer
  end

  create_table "sucursals", :force => true do |t|
    t.column "nombre",        :string
    t.column "gerente",       :string
    t.column "telefono",      :string
    t.column "direccion",     :string
    t.column "codigo_postal", :string
    t.column "municipio_id",  :integer
  end

  create_table "systables", :force => true do |t|
    t.column "controller_id", :integer
    t.column "rol_id",        :integer
    t.column "consultar",     :integer, :default => 1
    t.column "actualizar",    :integer, :default => 0
    t.column "insertar",      :integer, :default => 1
    t.column "eliminar",      :integer, :default => 1
  end

  create_table "tipo_transaccions", :force => true do |t|
    t.column "descripcion", :string
    t.column "prioridad",   :integer
  end

  create_table "transaccions", :force => true do |t|
    t.column "monto",                 :float
    t.column "pagogrupal_id",         :integer
    t.column "tipo_transaccion_id",   :integer
    t.column "fecha_hora_aplicacion", :datetime
    t.column "datafile_id",           :integer
  end

  create_table "transferencias", :force => true do |t|
    t.column "origen_id",     :integer
    t.column "destino_id",    :integer
    t.column "user_id",       :integer
    t.column "fecha",         :date
    t.column "monto",         :string
    t.column "observaciones", :string
    t.column "st",            :integer
  end

  create_table "ubicacion_negocios", :force => true do |t|
    t.column "ubicacion", :string
  end

  create_table "users", :force => true do |t|
    t.column "login",    :string
    t.column "password", :string
    t.column "nombre",   :string
    t.column "rol_id",   :integer, :default => 1
    t.column "st",       :integer
  end

  create_table "viviendas", :force => true do |t|
    t.column "tipo_vivienda", :string
  end

end
